Revamped code due to global variable issues etc

shahaadi opened the pull request:
Was trying to port to safari but had a ton of issues related to the safari webkit's inability to throw errors automatically in many cases that chrome/firefox does, so revamped a good deal of the code to remove errors caused by said issues + global variable conflict.

also renamed many variables to make it more clear what they represent and standardized the variable names across the same contexts in different files.

replaced xml with fetch

replaced bookmark logic with ISO formatting for easier storage/conversion (internally it all just checks seconds since jan 1 1970 @ 12:00 am before converting to month/day/year format)

changed html files to match HTML5 formatting

created new file for bookmark.html css separately

removed redundant calls to sendMessage so that I could wrap everything in asyncs (so that, once again, global variable conflicts do not occur)

standardized style across all files for readability

added error throwing/catching to make bug fixing easier

fixed issue of Entire work button on dark mode causing the text to turn black (while the background was dark gray), subsequently broke this fix but likely because i accidentally deleted a line or something

Made the stylistic choice to hide the Entire Work button after it is clicked (thus preventing another bug that destroys the page if you click Entire Work more than once).

Made the stylistic choice to hide the next and previous buttons after Entire Work is clicked (what chapters are you going to if they're all on the current page?)

Future plans:

- make word count show for each chapter next to the headings (Chapter 1, Chapter 2, etc) on the Entire Work page
- fix black text conversion after entire work clicked on dark mode
- Figure out why the console throws a "Failed to apply UI enhancements and features on profile page: TypeError: Cannot read properties of null (reading 'after')" error
- Sleep


PLEASE PLEASE PLEASE ACCEPT THESE CHANGES I DIDNT BREAK ANYTHING THAT WASNT ALREADY BROKEN

more importantly i think im well on the way to making this safari compatible

Waland2 commented:
Thank you so much for all the hard work - I really appreciate the effort you put into this!
I also noticed that old bookmarks were broken due to the fields name changes, and I've fixed it in the latest commit.

shahaadi commented:
i realized this today while i was testing my imported json on it, sorry about that; i think it was bc i changed the names for some variables! thanks for fixing!

i did include some backwards compatibility but i could include it into your change so that it automatically converts the old date format into the new one? 

also thank you for merging! i had a lot of fun working on it (and it now runs on safari, albeit with a few bugs).

i figured out why i broke some of the colors after clicking "Entire Work" - it was bc I changed the format for colors to hex when the native format on ff.net that the extension checks for is rgb, and for some reason those are never considered the same even if they have the same color value. Will fix in a later commit.

im well into the process of making modules for redundancy - i think honestly it may be possible to have a single file that runs on every single page (lists, crossovers, search, etc etc) and literally attempts to run all of the code but with null checks so if something doesn't exist it just doesn't run. i believe it would lose a massive amount of redundancy (i'm in the process of attempting to do this and the only file i have to consolidate into that universal file [from pages/] is story.js, with the module file only being about 500 lines long).
